tools/src/icon4pytools/py2fgen/wrappers/common.py

```diff
@@ -132,12 +136,12 @@ def construct_icon_grid(
 
 
 def construct_decomposition(
-    c_glb_index: xp.ndarray,
-    e_glb_index: xp.ndarray,
-    v_glb_index: xp.ndarray,
-    c_owner_mask: xp.ndarray,
-    e_owner_mask: xp.ndarray,
-    v_owner_mask: xp.ndarray,
+    c_glb_index: NDArray,
```

Comment:
```suggestion
    c_glb_index: xp.ndarray,
```
Same here you have the xp import from settings here.